kube-client/src/client/auth/mod.rs

```diff
@@ -146,30 +196,39 @@ impl RefreshableToken {
                         }
 
                         // Unreachable because the token source does not change
+                        Auth::RefreshableToken(RefreshableToken::File(_)) => unreachable!(),
                         #[cfg(feature = "oauth")]
                         Auth::RefreshableToken(RefreshableToken::GcpOauth(_)) => unreachable!(),
                     }
                 }
 
-                let mut value = HeaderValue::try_from(format!("Bearer {}", &locked_data.0.expose_secret()))
-                    .map_err(Error::InvalidBearerToken)?;
-                value.set_sensitive(true);
-                Ok(value)
+                bearer_header(locked_data.0.expose_secret())
+            }
+
+            RefreshableToken::File(token_file) => {
+                if let Some(token) = token_file.read().await.cached_token() {
+                    bearer_header(token)
+                } else {
+                    bearer_header(token_file.write().await.token())
+                }
```

Comment:
since this is an `RwLock`, is there a chance here that two calls started simultaneously while the token is expiring here would trigger two refreshes?

Reply:
i guess "triggering a refresh" here is really just reading the file again, one after the other, so it's probably not a big deal.

Reply:
> is there a chance here that two calls started simultaneously while the token is expiring here would trigger two refreshes?

There shouldn't be. `token()` method requires exclusive access, and only reloads if `is_expiring()`. If another call had reloaded the token while waiting for the exclusive access, calling `token()` again won't read the file.